db/flush_job.cc

```diff
@@ -850,20 +852,22 @@ Status FlushJob::WriteLevel0Table() {
   const uint64_t start_cpu_micros = clock_->CPUMicros();
   Status s;
 
-  SequenceNumber smallest_seqno = mems_.front()->GetEarliestSequenceNumber();
-  if (!db_impl_seqno_to_time_mapping_.Empty()) {
-    // make a local copy, as the seqno_to_time_mapping from db_impl is not
-    // thread safe, which will be used while not holding the db_mutex.
-    seqno_to_time_mapping_ =
-        db_impl_seqno_to_time_mapping_.Copy(smallest_seqno);
-  }
-
   std::vector<BlobFileAddition> blob_file_additions;
 
   {
     auto write_hint = cfd_->CalculateSSTWriteHint(0);
     Env::IOPriority io_priority = GetRateLimiterPriorityForWrite();
     db_mutex_->Unlock();
+    SequenceNumber smallest_seqno = mems_.front()->GetEarliestSequenceNumber();
+    {
+      std::shared_lock seqno_time_lock(*db_impl_seqno_to_time_mutex_);
```

Comment:
It might seem minor, but I would bet that this extra mutex acquire/release on every flush, even without the preserve/preclude option set, will cause a measurable performance regression on write-heavy workloads. :-/